### PR TITLE
Switch to registry.k8s.io for the sandbox Image:

### PR DESCRIPTION
**Pull request by ameukam**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Related:
  - https://github.com/kubernetes/k8s.io/issues/3411

The Kubernetes project setup a new endpoint that will used to service
the container images produced by the project.

#### Does this PR introduce a user-facing change?

```release-note
Updated default pause image to point to new location `registry.k8s.io/pause:3.6`
```

### CODE REVIEW

**Comment by haircommander** [comment]
thanks @ameukam !
/approve

File: docs/crio.8.md [hunk 1]
```diff
@@ -449,4 +449,4 @@ Shows a list of commands or help for one command
 # SEE ALSO
 
 crio.conf(5), crio.conf.d(5), oci-hooks(5), policy.json(5), registries.conf(5),
-storage.conf(5)
\ No newline at end of file
+storage.conf(5)
```
Comment (saschagrunert) [inline docs/crio.8.md]: We have to remove this change to make our CI happy :)

**Review by saschagrunert — COMMENTED** [review]
Code LGTM, thank you @ameukam!

The integration tests seems to fail on other PRs as well, which is something we have to investigate on our side.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by TomSweeneyRedHat** [comment]
LGTM
if the tests will cooperate

**Comment by rphillips** [comment]
Should we pick this into the current release branches?

**Comment by haircommander** [comment]
yeah probably, especially if registry.k8s.io support is ever dropped

/cherry-pick release-1.23

**Pull request merged**